semgrep/semgrep/rule_match.py

```diff
@@ -63,6 +65,18 @@ def start(self) -> Dict[str, Any]:
     def end(self) -> Dict[str, Any]:
         return self._end
 
+    @property
+    def lines(self) -> Iterable[str]:
+        """
+            Return lines in file that this RuleMatch is referring to.
+
+            Assumes file exists.  Note that start/end line is one-indexed
+        """
+        with self.path.open(
+            buffering=1, errors="replace"
+        ) as fin:  # buffering=1 turns on line-level reads
+            return list(itertools.islice(fin, self.start["line"] - 1, self.end["line"]))
```

Comment:
would we ever expect to be able to call `self.start.line`?

Reply:
Yeah start and end should be objects with named fields as well